### ISSUE TITLE
Data preparation for TTA example

### ISSUE THREAD

**Issue by zengchang233** [open-mmlab/Amphion#50]
Hi, thanks for your nice work. 

Could you provide the script for extracting the acoustic feature for the TTA task?

The beginner recipe said there are four stages for this task.
- Data preparation
- Train VAE
- Train LDM model
- Inference

I can find three scripts except the one for data preparation.

**Comment by HeCheng0625** [comment]
Sure, We will provide the script recently, and we will provide the processed data for AudioCaps (or more), you can directly download it.

**Comment by NoFish-528** [comment]
And there are one very small mistake( thought it's unnecessary, I think you can fix it easily
![image](https://github.com/open-mmlab/Amphion/assets/73390819/0909f271-4ae2-480a-8927-bf840ce0e542)

**Comment by RMSnow** [comment]
> And there are one very small mistake( thought it's unnecessary, I think you can fix it easily ![image](https://private-user-images.githubusercontent.com/73390819/292375348-0909f271-4ae2-480a-8927-bf840ce0e542.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMyMzI1NDQsIm5iZiI6MTcwMzIzMjI0NCwicGF0aCI6Ii83MzM5MDgxOS8yOTIzNzUzNDgtMDkwOWYyNzEtNGFlMi00ODBhLTg5MjctYmY4NDBjZTBlNTQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIyVDA4MDQwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzODNkZjE0ODMyOWNiYjdjNmFhZGUyY2FlZjdkMzkzOWJiNjA2OGU5MmE1ZGUxM2EzYTNkMmJhNTY1ZmRiMmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Bghfel6emQWdhpDma2ZCPxsng8bJ2tvbI-0Nq4fG3Ew)

@NoFish-528 Thanks for pointing that! @HeCheng0625 Please follow up with this

**Comment by ostinsolo** [comment]
Hi, can you give us some direction of how to use pre-trained models in the TTA or TTM recipe. How do we include these pre-trained make-an-audio (https://drive.google.com/drive/folders/1zZTI3-nHrUIywKFqwxlFO6PjB66JA8jI) or this (https://huggingface.co/cvssp/audioldm2) AudioLDM ?

**Comment by HarryHe11** [comment]
> Hi, can you give us some direction of how to use pre-trained models in the TTA or TTM recipe. How do we include these pre-trained make-an-audio (https://drive.google.com/drive/folders/1zZTI3-nHrUIywKFqwxlFO6PjB66JA8jI) or this (https://huggingface.co/cvssp/audioldm2) AudioLDM ?

@HeCheng0625 could you help with this issue please?

**Comment by HeCheng0625** [comment]
> Hi, can you give us some direction of how to use pre-trained models in the TTA or TTM recipe. How do we include these pre-trained make-an-audio (https://drive.google.com/drive/folders/1zZTI3-nHrUIywKFqwxlFO6PjB66JA8jI) or this (https://huggingface.co/cvssp/audioldm2) AudioLDM ?

Hi you can follow this recipe (https://github.com/open-mmlab/Amphion/blob/main/egs/tta/RECIPE.md) for how to using the pretrained models for text-to-audio.

And, you can download our pre-trained checkpoint in: https://huggingface.co/amphion/text_to_audio;
while, you can also use the online demo in: https://huggingface.co/spaces/amphion/Text-to-Audio

**Comment by HeCheng0625** [comment]
We will release AudioCaps dataset in huggingface in one week!

**Comment by jianrenw** [comment]
Thank you for your great work! Could you please provide more information about the data format? I am trying to encode a 10s 24000fps wav file into embedding space and decode it back. The decoded wav does not seem correct. Here is the code:

cfg = load_config("egs/tta/audioldm/exp_config.json")
autoencoderkl = build_autoencoderkl(cfg, device)
wav_path = 'my_audio.wav'
wav_torch, _ = audio.load_audio_torch(wav_path, 24000)
mel = extract_mel_features(wav_torch.unsqueeze(0), cfg.preprocess)
melspec = mel.unsqueeze(0).unsqueeze(1).to(device) 
posterior = autoencoderkl.encode(melspec)
latent = posterior.sample() 
mel_out = autoencoderkl.decode(latent)

melspec = mel_out[0,0].cpu().detach().numpy()

vocoder.eval()
vocoder.remove_weight_norm()

with torch.no_grad():
    melspec = np.expand_dims(melspec, 0)
    melspec = torch.FloatTensor(melspec).to(device)

    y = vocoder(melspec)
    audio = y.squeeze()
    audio = audio * 32768.0
    audio = audio.cpu().numpy().astype("int16")

os.makedirs("result", exist_ok=True)
write(os.path.join("result", 'shenqi' + ".wav"), 16000, audio)

**Comment by HeCheng0625** [comment]
> Thank you for your great work! Could you please provide more information about the data format? I am trying to encode a 10s 24000fps wav file into embedding space and decode it back. The decoded wav does not seem correct. Here is the code:
> 
> ```
> cfg = load_config("egs/tta/audioldm/exp_config.json")
> autoencoderkl = build_autoencoderkl(cfg, device)
> wav_path = 'my_audio.wav'
> wav_torch, _ = audio.load_audio_torch(wav_path, 24000)
> mel = extract_mel_features(wav_torch.unsqueeze(0), cfg.preprocess)
> melspec = mel.unsqueeze(0).unsqueeze(1).to(device) 
> posterior = autoencoderkl.encode(melspec)
> latent = posterior.sample() 
> mel_out = autoencoderkl.decode(latent)
> 
> melspec = mel_out[0,0].cpu().detach().numpy()
> 
> vocoder.eval()
> vocoder.remove_weight_norm()
> 
> with torch.no_grad():
>     melspec = np.expand_dims(melspec, 0)
>     melspec = torch.FloatTensor(melspec).to(device)
> 
>     y = vocoder(melspec)
>     audio = y.squeeze()
>     audio = audio * 32768.0
>     audio = audio.cpu().numpy().astype("int16")
> 
> os.makedirs("result", exist_ok=True)
> write(os.path.join("result", 'shenqi' + ".wav"), 16000, audio)
> ```

Hi, the wav should be sampled as 16KHz.

**Comment by HeCheng0625** [comment]
We update processed AudioCaps dataset: https://openxlab.org.cn/datasets/Amphion/AudioCaps

**Comment by HIN0209** [comment]
Hi, the URL below provides wav and npy files, but I cannot find train.json and valid.json. Are we supposed to create by us, but no caption list is available? 
https://openxlab.org.cn/datasets/Amphion/AudioCaps/tree/main
